### ISSUE TITLE
False alarm ternary operator which prints a plain string

### ISSUE THREAD

**Issue by capbussat** [WordPress/WordPress-Coding-Standards#2209]
## Bug Description

I think it's a false alarm in a ternary operator that prints a simple plain string.

## Minimal Code Snippet
The issue happens when running this command from Visual Studio Code terminal:
```bash
phpcs --standard=WordPress 
```
... over a file containing this code:
```php
function a() {
		$key = 'auto';
		( 'auto' === $key ) ? print ' disabled ' : print ' enabled '; // false alarm.
		( 'auto' === $key ) ? print esc_html( ' disabled ' ) : print esc_html( ' enabled ' );
}
```
## Error Code
`All output should be run through an escaping function (see the Security sections in the WordPress Developer Handbooks), found ':'.  (WordPress.Security.EscapeOutput.OutputNotEscaped)`

## Environment
PHP 7.4.33 (cli) (built: Nov  2 2022 16:00:55) ( ZTS Visual C++ 2017 x64 )
PHP_CodeSniffer version 3.7.1 (stable) by Squiz (http://www.squiz.net)
WPCS

**Comment by dingo-d** [comment]
@capbussat can you update the issue with the example and some more details?

**Comment by jrfnl** [comment]
@dingo-d Going by the mail I got, the info was already there, just hidden as an HTML comment. I've just edited the original post to remove the HTML comment tag and fix the code highlighting (no content edits), so you should now be able to see the report.

**Comment by jrfnl** [comment]
@capbussat Marking this as a duplicate as there are a few other issues open about the `EscapeOutput` sniff and ternary handling already. See: https://github.com/WordPress/WordPress-Coding-Standards/issues?q=is%3Aissue+is%3Aopen+ternary

The problem with ternaries is that it is often hard to know the start/end of a "leaf" of  the ternary, especially when things get nested etc.

All the same, I totally agree that for the code sample above, these are false positives and should be fixable.
I'm leaving the ticket open to make sure that the code sample can be found when this issue will be addressed in the `EscapeOutput` sniff.

**Comment by zizzfizzix** [comment]
I just ran across this with lines like these:
```
( isset( $this->tracking_for_divi_options['send_gads_conversion'] ) && 'on' === $this->tracking_for_divi_options['send_gads_conversion'] ) ? 'required' : ''
```
The error message is:

> All output should be run through an escaping function (see the Security sections in the WordPress Developer Handbooks), found 'isset'. (WordPress.Security.EscapeOutput.OutputNotEscaped)

This clearly is a false positive as `isset()` can only return a bool so escaping it is pointless.

**Comment by jrfnl** [comment]
@zizzfizzix Your issue is a different one, so please open a separate ticket. Also, the code sample you provided as-is, would not be flagged as there is no output being generated, so as this are, there is nothing actionable in your report.
